File: cpp/163248.cpp
typedef long long ll;

long long minCostRepairingCar(std::vector<int> a, std::vector<int> b) {
    int n = a.size();
    int cntMinA = 0;
    ll sum = 0ll;
    for (int i = 0; i < n; i++) {
        if (a[i] <= b[i]) {
            sum += a[i];
            cntMinA++;
        } else {
            sum += b[i];
        }
    }
    if (n == 1) return sum;
    if (cntMinA > 0 && cntMinA < n)
        return sum;
    ll res = (ll)1e9 * n;
    for (int i = 0; i < n; i++) {
        if (a[i] <= b[i]) {
            res = min(res, sum - a[i] + b[i]);
        }
    }
    return res;
}

File: py/163248.py
# lời giải từ bạn kien_the_sun - ngắn gọn :))

def minCostRepairingCar(a, b): 
    if len(a) == len(b) == 1:
       return min(a[0], b[0])
    ans = sum(min(i,j) for i,j in zip(a,b)) 
    if all(i < j for i,j in zip(a,b)):
       ans += min(j-i for i,j in zip(a,b)) 
    elif all(i > j for i,j in zip(a,b)):
       ans += min(i-j for i,j in zip(a,b))
    return ans